test/performance/perf_test.lua

```diff
@@ -234,6 +251,200 @@ local select_prepare = function(g)
+end
+
+local vshard_prepare = function(g)
+    g.router:eval([[
+        local vshard = require('vshard')
+
+        local function _vshard_insert(space_name, tuple)
+            local bucket_id = vshard.router.bucket_id_strcrc32(tuple[1])
+            return vshard.router.callrw(
+                bucket_id,
+                '_vshard_insert_storage',
+                { space_name, tuple, bucket_id }
+            )
+        end
+
+        rawset(_G, '_vshard_insert', _vshard_insert)
+
+
+        local function _vshard_select(space_name, key)
+            local bucket_id = vshard.router.bucket_id_strcrc32(key)
+            return vshard.router.callrw(
+                bucket_id,
+                '_vshard_select_storage',
+                { space_name, key }
+            )
+        end
+
+        rawset(_G, '_vshard_select', _vshard_select)
+
+
+        local function sort(a, b)
+            return a[1] < b[1]
+        end
+
+        local function _vshard_select_gt(space, key, opts)
+            assert(type(opts.limit) == 'number')
+            assert(opts.limit > 0)
+
+            local tuples = {}
+
+            for id, replicaset in pairs(vshard.router.routeall()) do
+                local resp, err = replicaset:call(
+                    '_vshard_select_storage',
+                    { space, key, nil, box.index.GT, opts.limit }
+                )
+                if err ~= nil then
+                    error(err)
+                end
+
+                for _, v in ipairs(resp) do
+                    table.insert(tuples, v)
+                end
+
+            end
+
+            -- Naive merger.
+            local response = { }
+
+            table.sort(tuples, sort)
+
+            for i = 1, opts.limit do
+                response[i] = tuples[i]
+            end
+
+            return response
+        end
+
+        rawset(_G, '_vshard_select_gt', _vshard_select_gt)
+
+
+        local function _vshard_select_secondary(space_name, index_name, key, opts)
+            assert(type(opts.limit) == 'number')
+            assert(opts.limit > 0)
+
+            local storage_response = {}
+
+            for id, replicaset in pairs(vshard.router.routeall()) do
+                local resp, err = replicaset:call(
+                    '_vshard_select_storage',
+                    { space_name, key, index_name, box.index.EQ, opts.limit }
+                )
+                if err ~= nil then
+                    error(err)
+                end
+
+                storage_response[id] = resp
+            end
+
+            -- Naive merger.
+            local response = { }
+
+            local ind = 0
+            for i = 1, opts.limit do
+                for _, tuples in pairs(storage_response) do
+                    if tuples[i] ~= nil then
+                        response[ind] = tuples[i]
+                    end
+                end
+
+                if ind == opts.limit then
+                    break
+                end
+            end
```

Comment:
`idx` is always zero? The naive merger does not sort the tuples?

Reply:
Maybe it worth to add reference results for test cases and check the results on a warm up (several first request, where we don't count timings).

Reply:
> `idx` is always zero?

I don't get the question. I take top records from each instance response until I reach expected count.

> The naive merger does not sort the tuples?

Yeah. Honestly, I have no idea how to work with merger and it seemed to me that introducing complicated tuples sorting here is not that easy and will require many additional lines of code.

Reply:
> I don't get the question. I take top records from each instance response until I reach expected count.

The response will always contain one tuple (in `response[0]`). It does not look as expected result.

> Yeah. Honestly, I have no idea how to work with merger and it seemed to me that introducing complicated tuples sorting here is not that easy and will require many additional lines of code.

The `table.sort()` accepts the comparator function. It is easy to implement, when you know a field number.

Reply:
> The response will always contain one tuple (in `response[0]`). It does not look as expected result.

Oh, now I get it. My bad.

Reply:
> The `table.sort()` accepts the comparator function. It is easy to implement, when you know a field number.

Ok, I have thought that something more complicated is required.

Reply:
> `idx` is always zero? The naive merger does not sort the tuples?

Resolved

Reply:
> Maybe it worth to add reference results for test cases and check the results on a warm up (several first request, where we don't count timings).

It sounds like a relevant comment, but also as one that requires refactoring and additional work. I think we skip this one for now